### PR TITLE
KOGITO-4341 Infinispan upgrade to version 12

### PR DESCRIPTION
**Pull request by cristianonicolai**
Assembly:
- https://github.com/kiegroup/kogito-runtimes/pull/1031
- https://github.com/kiegroup/kogito-runtimes/pull/1341
- https://github.com/kiegroup/kogito-apps/pull/815
- https://github.com/kiegroup/kogito-examples/pull/692

### CODE REVIEW

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

File: kogito-test-utils/README.md [hunk 1]
```diff
@@ -24,15 +24,6 @@ public class MyTest {
 }
 ```
 
-And add the Infinispan properties in the _application.properties_:
-
-```
-#Infinispan
-quarkus.infinispan-client.use-auth=true
-quarkus.infinispan-client.auth-username=admin
-quarkus.infinispan-client.auth-password=admin
```
Comment (MarianMacik) [inline kogito-test-utils/README.md:33]: So Infinispan 12 now allows running without any authentication like it was with 10 if I remember correctly? Shouldn't we test it rather with authentication as this is more real-world use case? Or maybe we can leave that to Openshift BDD tests...

Comment (cristianonicolai) [reply kogito-test-utils/README.md:33]: agree that testing with auth is more realistic, but that has been causing issues during the ISPN upgrades. Decided to rollback to make future work easier and at the end of the day auth should be already a validated thing from the client anyway. We can revert this later on if we find the need to.

File: kogito-test-utils/src/main/java/org/kie/kogito/testcontainers/springboot/InfinispanSpringBootTestResource.java [hunk 2]
```diff
@@ -31,8 +34,11 @@ public InfinispanSpringBootTestResource() {
     }
 
     @Override
-    protected String getKogitoProperty() {
-        return KOGITO_INFINISPAN_PROPERTY;
+    protected Map<String, String> getProperties() {
+        Map<String, String> properties = new HashMap<>();
+        properties.put(KOGITO_INFINISPAN_PROPERTY, "localhost:" + getTestResource().getMappedPort());
+        properties.put("quarkus.infinispan-client.use-auth", "false");
```
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply kogito-test-utils/src/main/java/org/kie/kogito/testcontainers/springboot/InfinispanSpringBootTestResource.java:40]: defiitely

File: kogito-test-utils/src/test/java/org/kie/kogito/testcontainers/quarkus/InfinispanQuarkusTestResourceTest.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public void setup() {
     @Test
     public void shouldGetProperty() {
         givenResource();
-        assertEquals(InfinispanQuarkusTestResource.KOGITO_INFINISPAN_PROPERTY, resource.getKogitoProperty());
+        assertThrows(IllegalStateException.class, () -> resource.getProperties().get(InfinispanQuarkusTestResource.KOGITO_INFINISPAN_PROPERTY));
```
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply kogito-test-utils/src/test/java/org/kie/kogito/testcontainers/quarkus/InfinispanQuarkusTestResourceTest.java:42]: true, we should consider that.

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by evacchi — APPROVED** [review]

File: kogito-test-utils/README.md (hunk 1 above)
Comment (MarianMacik) [reply kogito-test-utils/README.md:33]: +1

**Review by MarianMacik — APPROVED** [review]

**Comment by cristianonicolai** [comment]
Jenkins rerun apps tests

**Comment by cristianonicolai** [comment]
Jenkins rerun examples tests

**Comment by cristianonicolai** [comment]
Jenkins rerun apps tests

**Comment by cristianonicolai** [comment]
Jenkins rerun apps tests

**Pull request merged**